src/autocomplete/AutocompleteView.js

```diff
@@ -46,7 +46,9 @@ export default function AutocompleteView(props: Props): Node {
 
   const { lastWordPrefix, filter } = getAutocompleteFilter(text, selection);
   const AutocompleteComponent = prefixToComponent[lastWordPrefix];
-  const shouldShow = isFocused && !!AutocompleteComponent && filter.length > 0;
+  const shouldShow =
+    (isFocused && !!AutocompleteComponent && filter.length > 0)
+    || (isFocused && !!AutocompleteComponent && lastWordPrefix === '@');
```

Comment:
I think this would look better after deduplicating it a little, and maybe adding some comments.

I also wonder if we can avoid repeating the string literal '@', and make it a bit clearer what's going on, by replacing the `lastWordPrefix === '@'` condition with `AutocompleteComponent === PeopleAutocomplete`?

Then maybe a comment linking to more discussion of why we want this particular behavior for the people autocomplete but not for the other autocompletes.

I'm thinking maybe something like this:

```js
  /* prettier-ignore */
  const shouldShow =
    isFocused
    && !!AutocompleteComponent
    && (
      // For most autocompletes, wait until the user has typed at least one
      // character of their search term…
      filter.length > 0
      // …but for the people autocomplete, just show it without waiting
      // (i.e., just after the user has typed '@'). See
      //   https://github.com/zulip/zulip-mobile/issues/4872#issuecomment-942612417.
      || AutocompleteComponent === PeopleAutocomplete
    );
```

I do have a vague feeling that there's some better place to put this logic. If so, maybe @gnprice will spot it? 🙂

Reply:
Thanks @chrisbobbe for the pointers, and will work on it now and ask @gnprice for ideas on where we could put the extra logic.